peps/pep-0750.rst

```diff
@@ -53,837 +33,1089 @@ deferred pending further experience with f-strings. Work on this PEP was
-
-Likewise, the inability to intercept interpolated values means that they cannot be
-sanitized or otherwise transformed before being integrated into the final string. Here,
-the convenience of f-strings could be considered a liability. For example, a user
-executing a query with `sqlite3 <https://docs.python.org/3/library/sqlite3.html>`__
-may be tempted to use an f-string to embed values into their SQL expression instead of
-using the ``?`` placeholder and passing the values as a tuple to avoid an
-`SQL injection attack <https://en.wikipedia.org/wiki/SQL_injection>`__.
-
-Tag strings address both these problems by extending the f-string syntax to provide
-developers access to the string and its interpolated values before they are combined. In
-doing so, tag strings may be interpreted in many different ways, opening up the
-possibility for DSLs and other custom string processing.
-
-Proposal
-========
-
-This PEP proposes customizable prefixes for f-strings. These f-strings then
-become a "tag string": an f-string with a "tag function." The tag function is
-a callable which is given a sequence of arguments for the parsed tokens in
-the string.
+Python f-strings are easy to use and very popular. Over time, however, developers
+have encountered limitations that make them unsuitable for certain use cases. 
+In particular, f-strings provide no way to intercept and transform interpolated 
+values before they are combined into a final string.
 
-Here's a very simple example. Imagine we want a certain kind of string with
-some custom business policies: uppercase the value and add an exclamation point.
-
-Let's start with a tag string which simply returns a static greeting:
-
-.. code-block:: python
+As a result, incautious use of f-strings can lead to security vulnerabilities. 
+For example, a user executing a SQL query with `sqlite3 <https://docs.python.org/3/library/sqlite3.html>`__
+may be tempted to use an f-string to embed values into their SQL expression 
+instead of using the ``?`` placeholder and passing the values as a tuple to 
+avoid a `SQL injection attack <https://en.wikipedia.org/wiki/SQL_injection>`__.
 
-    def greet(*args):
-        """Give a static greeting."""
-        return "Hello!"
+More broadly, the inability to transform interpolated values before they are
+combined into a final string limits the utility of f-strings in more complex
+string processing tasks. 
 
-    assert greet"Hi" == "Hello!"  # Use the custom "tag" on the string
+Template strings address these problems by providing 
+developers with access to the string and its interpolated values before they 
+are combined. 
 
-As you can see, ``greet`` is just a callable, in the place that the ``f``
-prefix would go. Let's look at the args:
+For example, imagine we want to generate some HTML. Using template 
+strings, we can define an ``html`` function that allows us to automatically 
+sanitize interpolated values:
```

Comment:
Seems a bit odd to pivot from SQL as the motivating example above to HTML here -- perhaps one could be changed?

Reply:
I wanted to give two distinct injection/XSS-style examples, but agree there should be better connecting content here.

Reply:
Took a stab at this. See how you like it.